contract/contracts/AstarBase.sol

```diff
@@ -31,6 +31,7 @@ contract AstarBase is Ownable {
     function register(bytes32 ss58PublicKey, bytes calldata signedMsg) external {
         require(!paused, "The contract is paused");
         require(ss58PublicKey != 0, "Can't register ss58PublicKey with 0");
+        require(ss58Map[ss58PublicKey] == address(0), "already used ss58 Public Key");
```

Comment:
are there any checks for `addressMap[evmAddress]`?

Reply:
let me fix this. There is no harm to not check, but it will leave one ss58Map entry in strange state